dev/breeze/src/airflow_breeze/commands/sbom_commands.py

```diff
@@ -291,25 +291,74 @@ def generate_providers_requirements(
         with open(PROVIDER_METADATA_JSON_FILE_PATH) as f:
             provider_metadata = json.load(f)
             providers_info = [
-                (p_id, p_version, info["associated_airflow_version"])
-                for (p_id, p_versions) in provider_metadata.items()
-                for (p_version, info) in p_versions.items()
+                (
+                    provider_id,
+                    provider_version,
+                    python_version,
+                    info["associated_airflow_version"],
+                )
+                for (provider_id, provider_versions) in provider_metadata.items()
+                for (provider_version, info) in provider_versions.items()
+                for python_version in AIRFLOW_PYTHON_COMPATIBILITY_MATRIX[info["associated_airflow_version"]]
+                if python_version in python_versions
             ]
     else:
         if airflow_version is None:
             airflow_version = get_active_airflow_versions(confirm=False)[-1]
             get_console().print(f"[info]Using {airflow_version} as airflow version")
-        providers_info = [(provider_id, None, airflow_version) for provider_id in provider_ids]
+        providers_info = [
+            (provider_id, None, python_version, airflow_version)
+            for provider_id in provider_ids
+            for python_version in AIRFLOW_PYTHON_COMPATIBILITY_MATRIX[airflow_version]
+            if python_version in python_versions
+        ]
 
-    build_all_airflow_versions_base_image(python_version=python)
+    if run_in_parallel:
```

Comment:
It would be nice to have `--build-images` flag added and only run this one when the flag is set. This way we will skip even checking if the images are build.

Reply:
Actually - probably even better if we separate it out to a separte command I think

* breeze sbom build-all-airflow-image
* breeze sbom generate-provider-requirements

Reply:
This way we could separately build either the image for "all versions" (if we need to regenerate sbom information for historical versions of providers for whatever reason separately. 

And then - we could choose the "all-airflow" images instead of the "last" airflow" image for that.

Reply:
Done, we have now a separate command to build all airflow images. `breeze sbom build-all-airflow-images`